botcity/maestro/sdk.py

```diff
@@ -1067,6 +1070,50 @@ def create_credential(self, label: str, key: str, value):
             if not req.ok:
                 req.raise_for_status()
 
+    @since_version("3.0.0")
+    @ensure_access_token()
+    def update_credential(self, label: str, key: str, new_value):
+        """Update a credential with a new value
+        Args:
+            label: Credential set name
+            key: Key name within the credential set
+            new_value: Credential new value
+        """
+        credential = self._get_credential_by_label(label=label)
+        if credential is None:
+            raise ValueError(f"Credential set '{label}' does not exists.")
+        secrets = credential.get("secrets", [])
+
+        index = next((i for i, d in enumerate(secrets) if d.get('key') == key), None)
+        if index is None:
+            raise ValueError(f"Key '{key}' does not exist in credential set '{label}'.")
+        else:
+            secrets[index]['value'] = new_value
+        response = self._edit_credential_by_label(label, secrets=secrets)
+        if response is None:
+            raise ValueError('Error during credential update.')
```

Comment:
Could we adjust the _edit_credential_by_label function to properly raise the desired error, making future maintenance easier? What do you think?

The modification would be almost the same as the one I implemented in the get example.

Reply:
Addressed via https://github.com/botcity-dev/botcity-maestro-sdk-python/pull/41/commits/522ebc5d4fa3b58b0acc309ffa6237c4ff03a001